File: compountviewjavademo/src/main/java/com/example/compountviewjavademo/MainActivity.java
package com.example.compountviewjavademo;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.LinearLayout;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViewById(R.id.btnAdd).setOnClickListener(this::onClick);

        final LayoutInflater inflate2=getLayoutInflater();
        inflate2.inflate(R.layout.compound_view,(LinearLayout)findViewById(R.id.layoutCompound),true);

    }

    private void onClick(View view) {
        //The LayoutInflater you can declared in three ways....

        //First way...
        // final LayoutInflater inflater=(getLayoutInflater()getSystemService(LAYOUT_INFLATER_SERVICE));

        //Second way to creat Inflater below..
        //final LayoutInflater inflater1=LayoutInflater.from(this);

        //Third way using Activity class gives you something called as getLayoutInflater..
        final LayoutInflater inflate2=getLayoutInflater();
        inflate2.inflate(R.layout.compound_view,(LinearLayout)findViewById(R.id.layoutCompound),true);

    }
}

File: app/src/main/java/com/example/pc/compoundviewdemo/MainActivity.kt
package com.example.pc.compoundviewdemo

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.layout_compound.view.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btnOk.setOnClickListener(this::Clicked)

    }

    private fun Clicked(view: View) {
        val vm: View = layoutInflater.inflate(R.layout.layout_compound, null, false)
        vm.findViewById<ImageView>(R.id.imgVw).setOnClickListener {
            (it as ImageView).setImageResource(R.mipmap.ic_launcher_round)
        }

        vm.findViewById<TextView>(R.id.txtVw).setOnClickListener {
            (
                    it as TextView).textSize = 40f
        }
        layoutRoot.addView(vm)
    }
}
    /**private fun Clicked(view:View){
    val vm: View = layoutInflater.inflate(R.layout.layout_compound, null, false)
    vm.findViewById<ImageView>(R.id.imgVw1).setOnClickListener
    {
        (it as ImageView).setImageResource(R.mipmap.ic_launcher)
    }

    vm.findViewById<TextView>(R.id.txtVw1).setOnClickListener
    {
        (it as TextView).textSize = 50f
    }
    vm.findViewById<ImageView>(R.id.imgVw2).setOnClickListener
    {
        (it as ImageView).setImageResource(R.mipmap.ic_launcher)
    }
    vm.findViewById<TextView>(R.id.txtVw2).setOnClickListener
    {
        (it as TextView).textSize = 50f
    }

}
*/

